libpod/runtime_img.go

```diff
@@ -1262,3 +1266,219 @@ func getPolicyContext(ctx *types.SystemContext) (*signature.PolicyContext, error
+func (r *Runtime) GetImagesForImages() ([]inspect.ImageResult, error) {
+	var results []inspect.ImageResult
+
+	foo, err := r.store.Images()
+	if err != nil {
+		return nil, err
+	}
+	for _, image := range foo {
+		bar, err := is.Transport.ParseStoreReference(r.store, image.ID)
+		if err != nil {
+			return nil, err
+		}
+		systemContext := &types.SystemContext{}
+		img, err := bar.NewImageSource(systemContext)
+		if err != nil {
+			return nil, err
+		}
+		ic, err := bar.NewImage(&types.SystemContext{})
+		if err != nil {
+			return nil, err
+		}
+		imgInspect, err := ic.Inspect()
+		if err != nil {
+			return nil, err
+		}
+		dangling := false
+		if len(image.Names) == 0 {
+			dangling = true
+		}
+
+		for repo, tag := range reposToMap(image.Names) {
+			results = append(results, inspect.ImageResult{
+				ID:         image.ID,
+				Repository: repo,
+				RepoTags:   image.Names,
+				Tag:        tag,
+				//RepoDigests:,
+				Size:   imageSize(img),
+				Digest: image.Digest,
+				//ConfigDigest:,
+				Created:  image.Created,
+				Labels:   imgInspect.Labels,
+				Dangling: dangling,
+			})
+		}
+
+	}
+	return results, nil
+}
+
+func imageCreatedBefore(createTime time.Time) ImageResultFilter {
+	return func(i inspect.ImageResult) bool {
+		if i.Created.Before(createTime) {
+			return true
+		}
+		return false
+	}
+}
+
+func imageCreatedAfter(createTime time.Time) ImageResultFilter {
+	return func(i inspect.ImageResult) bool {
+		if i.Created.After(createTime) {
+			return true
+		}
+		return false
+	}
+}
+
+func imageDangling() ImageResultFilter {
+	return func(i inspect.ImageResult) bool {
+		if i.Dangling {
+			return true
+		}
+		return false
+	}
+}
+
+func imageLabel(labelfilter string) ImageResultFilter {
+	// We need to handle both label=key and label=key=value
+	return func(i inspect.ImageResult) bool {
+		var value string
+		splitFilter := strings.Split(labelfilter, "=")
+		key := splitFilter[0]
+		if len(splitFilter) > 1 {
+			value = splitFilter[1]
+		}
+		for labelKey, labelValue := range i.Labels {
+			// handles label=key
+			if key == labelKey && len(strings.TrimSpace(value)) == 0 {
+				return true
+			}
+			//handles label=key=value
+			if key == labelKey && value == labelValue {
+				return true
+			}
+		}
+		return false
+	}
+}
+
+func imageFilter(name string) ImageResultFilter {
+	return func(i inspect.ImageResult) bool {
+		li := strings.LastIndex(name, ":")
+		var repository, tag string
+		if li < 0 {
+			repository = name
+		} else {
+			repository = name[0:li]
+			tag = name[li+1:]
+		}
+		if repository == i.Repository && len(strings.TrimSpace(tag)) == 0 {
+			return true
+		}
+		if repository == i.Repository && tag == i.Tag {
+			return true
+		}
+		return false
+	}
+}
+
+// CreateFilterFuncs returns an array of filter functions based on the user inputs
+// and is later used to filter images for output
+func CreateFilterFuncs(r *Runtime, c *cli.Context, userInput string) ([]ImageResultFilter, error) {
```

Comment:
Can you make the individual ImageResultFilter funcs public and throw this in cmd/podman?

Reply:
done